app/TuistApp/Sources/Views/DevicesView/DevicesViewModel.swift

```diff
@@ -222,6 +242,20 @@ final class DevicesViewModel: Sendable {
             try await deviceController.launchApp(bundleId: app.infoPlist.bundleId, device: device)
         }
     }
+
+    private func categorizeDevices() {
+        (connectedDevices, disconnectedDevices) = devices
+            .reduce(
+                into: (connected: [PhysicalDevice](), disconnected: [PhysicalDevice]())
+            ) { partialResult, device in
+                switch device.connectionState {
+                case .connected:
+                    partialResult.connected.append(device)
+                case .disconnected:
+                    partialResult.disconnected.append(device)
+                }
+            }
```

Comment:
Isn't this more of a filter operation?

We could even consider making these two properties computed variables:
```swift
var connectedDevices: [PhysicalDevice] {
  devices.filter { $0.connectionState == .connected }
}
```

While this approach is a bit less performant, we don't expect to have a large number of devices, so I think a better readability is a better tradeoff here.

Reply:
Thanks for your guidance! I made the changes based on your suggestions, the readonly property can update views in the `Observable`. Pretty cool!